src/algebra/ring/idempotents.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import algebra.ring.basic
+import algebra.group_power.basic
+import tactic.nth_rewrite.default
+
+/-!
+# Idempotents
+
+This file defines idempotents for an arbitary multiplication and proves some basic results:
+
+* In a semigroup, the product of two commuting idempotents is an idempotent;
```

Comment:
Can you add the full names of these results? Thanks!

Reply:
Done, thanks.